### DISCUSSION TITLE
Is it possible to do ERT in transimission with pygimli?

### DISCUSSION THREAD

**Discussion by Segnaly-2021** [gimli-org/pyGIMLi#540] [Q&A] [answered]
Hello everybody !

Here I'm, a little bit confused and asking for help and guidance.
I'm currently doing my final year internship and new to the geophysical world.
So maybe, the questions that i'm about to raise are obvious.

But, here is what I am struggling with? 

In my point of view, an electrical resistivity tomography (ERT) in transmission requires in the case for instance of a square surface putting current electrode on one side and the potential electrodes on the opposite side.

- All i can do with "ert.createData" is to choose the number of electrodes or their position and their configuration (dd, s, w, ...). So up to now, I don't know how to separately specify the positions of the current electrodes and receiver (potential electrodes). 

- Another thing that is not crucial but is important to me is to make electrodes visible on the geometry.

- If it possible, I would like to do something like the image below where green points are current electrodes and the red ones are potential electrodes. 

![image](https://github.com/gimli-org/gimli/assets/134691575/8b869129-4098-4072-8fac-a095c7c146a5)

Looking forward to hear from you!
Best regards,
Aly

**Comment by halbmy** [comment, marked as answer]
In geophysics, on either uses surface electrode layouts (for which createERTdata is made) or crosshole environments, in rare cases electrodes circumventing an object. For all of these cases there are examples on the pyGIMLi website:
- synthetic or field surface ERT
- crosshole ERT
- circumventing ERT (in joint inversion)

Usually, one uses a multielectrode system that can use any two (!) electrodes for current injection and any two others for potential measurement. So there is no need (and no sense) to subdivide the electrodes into current or potential ones.

**Reply by Segnaly-2021** [reply]
Hello, 

First, I'd like to thank you for this response and this amazing library.

And your response made me think that I need maybe to explain more what I want to do:

- I have a geometry which is constituted with two layers, a saturated one (with water) and a non-saturated.
- I want to follow the saturation front and therefore covering all the perimeter with electrodes even if current and potential electrodes will be kept together will give a better mapping.
- the electrodes will be evenly spaced (5cm) as shown in this image below
 
![image](https://github.com/gimli-org/gimli/assets/134691575/f6892a76-5d18-4f94-a8fd-956fc5c692c0)

Does it really make sense to you?

Following the 2D-ERT-modeling-and-inversion example, I've been trying to do first something simple by putting only 16 electrodes on the left side of the geometry(image above). But the pseudo-section that I have is a little bit confusing in the sense that I don't understand how the electrodes are moved to generating synthetic data?  

Is it a profiling or a vertical-electrical sounding or a combination of both? 
In the image below, what is the horizontal axis, depth of investigation(DOI)? Because I don't understand why I have only 14 and not something close to the length of electrode line(90)?

![image](https://github.com/gimli-org/gimli/assets/134691575/b0a4c44c-41c7-4cb8-9b72-d0179b98d4c5)

**Comment by halbmy** [comment]
A pseudosection is a combination of a profiling (horizontal axis) and depth sounding (vertical axis). In your case the profile is vertical but still the horizontal axis is along your electrode line. Just as the algorithm sees that the x values are not increasing as typical for surface ERT, it uses the electrode number instead of any (x, or z or what) position. Therefore the first midpoint is 1.5 (0-1-2-3 between 1 and 2) and so on. The vertical axis means increasing depth (in your case side) penetration but you cannot easily write a depth as it is an integrated measurement. Just consider the pseudosection a coloured data table.

Another comment: better use metres instead of centimeters for the geometric scaling, otherwise the modelling result will be Ohm-cm.

**Comment by halbmy** [comment]
From v1.4.7 on, there is one example of ERT around a tree: https://dev.pygimli.org/_examples_auto/3_ert/plot_06_ert_tree.html that might help. Furthermore, you can use `data.show()`/`ert.showData()`/`ert.show()` with `style='A-M'` etc. for non-line data display.
